src/Host/Client/Impl/Extensions/RStringExtensions.cs

```diff
@@ -17,7 +17,18 @@ public static class RStringExtensions {
                 return nullValue;
             }
 
-            return quote + s.Replace("\\", "\\\\").Replace("" + quote, "\\" + quote) + quote;
+            return quote + s.Replace("\\", "\\\\").Replace("" + quote, "\\" + quote).EscapeCharacterConstants() + quote;
+        }
+
+        private static string EscapeCharacterConstants(this string s) {
+            s = s.Replace("\n", "\\n")
```

Comment:
In settings textbox, the string with length 2 `\\n` becomes the R string literal `"\\\n"`, and when going back, the first backslash indicates escape, the second one is added to C# string.  Third backslash indicates escape, then 'n' means we add newline char to the C# string, which means textbox now has same original value `\\n`.

Without the change, we can't round-trip starting from settings.R.  If there's the R literal `"\t"` (length 1) it becomes just `t` in the settings textbox, and then when written back to settings.r, just `"t" again.